tests/integration/wireguard_test.py

```diff
@@ -0,0 +1,139 @@
+#
+# Copyright (c) 2021 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from contextlib import contextmanager
+
+
+import libnmstate
+from libnmstate.schema import Interface
+from libnmstate.schema import InterfaceState
+from libnmstate.schema import InterfaceType
+from libnmstate.schema import Wireguard
+
+from .testlib import assertlib
+
+
+WG0 = "wg0"
+WG0_PRIVATE_KEY = "exampleZp5x87cNoRJaHdAOzxrxDfDUn7pGmrY/AmzI="
+WG1_PRIVATE_KEY = "otherkeyp5x87cNoRJaHdAOzxrxDfDUn7pGmrY/AmzI="
+PEER0_PRESHARED_KEY = "exampleABe4wWyz4jh+uwX7vqRpNeGEtgAnUbwNjEug="
+PEER0_PUBLIC_KEY = "2Gl0SATbfrrzxfrSkhNoRR9Jg56y533y07KtIVngAk0="
+
+
+def test_add_and_remove_wireguard():
+    with wireguard_interface(WG0) as desired_state:
+        assertlib.assert_state(desired_state)
+
+    assertlib.assert_absent(WG0)
+
+
+def test_modify_wireguard():
+    with wireguard_interface(WG0) as desired_state:
+        wg_state = desired_state[Interface.KEY][0][Wireguard.CONFIG_SUBTREE]
+        wg_state[Wireguard.PRIVATE_KEY] = WG1_PRIVATE_KEY
+        wg_state[Wireguard.LISTEN_PORT] = 9500
+
+        libnmstate.apply(desired_state)
+        assertlib.assert_state(desired_state)
```

Comment:
Please use `assert_state_match`, in the future there might be more properties added to wireguard, the `assert_state` will fail then.